interface/src/app/download/download.spec.js

```diff
@@ -2,21 +2,113 @@
  * Unit tests.
  */
 
-describe('DownloadCtrl', function() {
-  describe('isCurrentUrl', function() {
-    var DownloadCtrl, $location, $scope;
-
-    beforeEach(module('adage'));
-
-    beforeEach(inject(function($controller, _$location_, $rootScope) {
-      $location = _$location_;
-      $scope = $rootScope.$new();
-      DownloadCtrl = $controller('DownloadCtrl',
-                                 {$location: $location, $scope: $scope});
-    }));
-
-    it('validates DownloadCtrl controller', inject(function() {
-      expect(DownloadCtrl).toBeTruthy();
-    }));
+// Test the generic controller properties (without server calls):
+describe('DownloadCtrl w/o server calls', function() {
+  beforeEach(module('adage.download'));
+
+  var ctrl, mockService;
+  // Create a mocked "AnnotationTypes" service, which overwrites the service
+  // in download.js (because it is loaded after "adage.download" module).
+  beforeEach(module(function($provide) {
+    mockService = {
+      get: function() { }
```

Comment:
<strike>Shouldn't the `mockService` be returning mocked-up data? Even if it's just a few lines, I think the purpose is to simulate the function of the service.</strike> Nevermind... I now see your comment below.